libvmi/driver/xen/xen_events.c

```diff
@@ -246,9 +249,10 @@ status_t process_interrupt_event(vmi_instance_t vmi,
         event->interrupt_event.gla = req->data.regs.x86.rip;
         event->interrupt_event.intr = intr;
         event->interrupt_event.reinject = -1;
-#if VM_EVENT_INTERFACE_VERSION >= 2
-        event->interrupt_event.insn_length = req->u.software_breakpoint.insn_length;
-#endif
+
+        if ( req->version >= 2 )
```

Comment:
Just a note for posterity: this modification is not related to support for VM_EVENT_FLAG_SET_REGISTERS, but updated in passing/while rebasing.